frontend/app/src/screens/diff/diff-tree.tsx

```diff
@@ -16,11 +18,19 @@ export default function DiffTree({ nodes, loading, ...props }: DiffTreeProps) {
   const navigate = useNavigate();
 
   useEffect(() => {
+    let rootModelMap: Record<string, Array<string>> = {};
+
     const formattedNodes: TreeProps["data"] = nodes.flatMap((node: DiffNode) => {
+      const currentModel = rootModelMap[node.kind];
+      rootModelMap = {
```

Comment:
is it possible to avoid variable re definition for a more functional way?

Reply:
tbh, I feel like mutation here is ok and explicit enough. I did some renaming update to make code a bit clearer